test/core/vm/actor/builtin/v2/market/market_actor_test.cpp

```diff
@@ -0,0 +1,574 @@
+  using primitives::cid::dataCommitmentV1ToCID;
+  using primitives::piece::PaddedPieceSize;
+  using primitives::piece::PieceInfo;
+  using primitives::sector::RegisteredSealProof;
+  using storage::ipfs::InMemoryDatastore;
+  using testing::_;
+  using testing::Return;
+  using testutil::vm::actor::builtin::ActorTestFixture;
+  using vm::runtime::MockRuntime;
+  using vm::state::StateTreeImpl;
+  using vm::version::NetworkVersion;
+  using namespace types::market;
+
+  const auto some_cid = "01000102ffff"_cid;
+  DealId deal_1_id = 13, deal_2_id = 24;
+
+  /// DealState cbor encoding
+  TEST(MarketActorCborTest, DealState) {
+    expectEncodeAndReencode(DealState{1, 2, 3}, "83010203"_unhex);
+  }
+
+  struct MarketActorTest : public ActorTestFixture<MarketActorState> {
+    void SetUp() override {
+      ActorTestFixture<MarketActorState>::SetUp();
+      ipld->load(state);
+      actorVersion = ActorVersion::kVersion2;
+
+      runtime.resolveAddressWith(state_tree);
+
+      currentEpochIs(50000);
+
+      addressCodeIdIs(miner_address, kStorageMinerCodeId);
+      addressCodeIdIs(owner_address, kAccountCodeId);
+      addressCodeIdIs(worker_address, kAccountCodeId);
+      addressCodeIdIs(client_address, kAccountCodeId);
+      addressCodeIdIs(kInitAddress, kInitCodeId);
+
+      EXPECT_CALL(*state_manager, createMarketActorState(testing::_))
+          .WillRepeatedly(testing::Invoke([&](auto) {
+            auto s = std::make_shared<MarketActorState>();
+            ipld->load(*s);
+            return std::static_pointer_cast<states::MarketActorState>(s);
+          }));
+
+      EXPECT_CALL(*state_manager, getMarketActorState())
+          .WillRepeatedly(testing::Invoke([&]() {
+            EXPECT_OUTCOME_TRUE(cid, ipld->setCbor(state));
+            EXPECT_OUTCOME_TRUE(current_state,
+                                ipld->getCbor<MarketActorState>(cid));
+            auto s = std::make_shared<MarketActorState>(current_state);
+            return std::static_pointer_cast<states::MarketActorState>(s);
+          }));
+    }
+
+    void expectSendFunds(const Address &address, TokenAmount amount) {
+      EXPECT_CALL(runtime, send(address, kSendMethodNumber, testing::_, amount))
+          .WillOnce(Return(outcome::success()));
+    }
+
+    void expectHasDeal(DealId deal_id, const DealProposal &deal, bool has) {
+      if (has) {
+        EXPECT_OUTCOME_EQ(state.proposals.get(deal_id), deal);
+      } else {
+        EXPECT_OUTCOME_EQ(state.proposals.has(deal_id), has);
+      }
+    }
+
+    ClientDealProposal setupPublishStorageDeals();
+
+    DealProposal setupVerifyDealsOnSectorProveCommit(
+        const std::function<void(DealProposal &)> &prepare);
+
+    Address miner_address{Address::makeFromId(100)};
+    Address owner_address{Address::makeFromId(101)};
+    Address worker_address{Address::makeFromId(102)};
+    Address client_address{Address::makeFromId(103)};
+
+    StateTreeImpl state_tree{ipld};
+  };
```

Comment:
I propose to add `MarketActorTestFixture : public ActorTestFixture<MarketActorState>` with all that stuff and inherit v0::MarketActor and v2::MarketActor from it.